src/pages/sessions/running.js

```diff
@@ -80,60 +93,72 @@ const Running = ({ intl, shortname }) => (
+                                        refetchQueries: [
+                                          { query: SessionListQuery },
+                                          { query: RunningSessionQuery },
+                                          { query: AccountSummaryQuery },
+                                        ],
+                                        variables: { id },
+                                      })
 
-                                  // redirect to the question pool
-                                  // TODO: redirect to a session summary or overview page
-                                  Router.push('/questions')
-                                }}
-                                handleNextBlock={() => {
-                                  activateNextBlock({
-                                    refetchQueries: [{ query: RunningSessionQuery }],
-                                  })
-                                }}
-                                handlePauseSession={async () => {
-                                  await pauseSession({
-                                    refetchQueries: [
-                                      { query: SessionListQuery },
-                                      { query: RunningSessionQuery },
-                                      { query: AccountSummaryQuery },
-                                    ],
-                                    variables: { id },
-                                  })
+                                      Router.push('/sessions')
+                                    }}
+                                    handleResetQuestionBlock={async () => {
+                                      await resetQuestionBlock({
+                                        variables: { id, instanceIds: activeInstanceIds },
+                                      })
 
-                                  Router.push('/sessions')
-                                }}
-                                handleTogglePublicEvaluation={() => {
-                                  updateSettings({
-                                    variables: {
-                                      sessionId: id,
-                                      settings: {
-                                        isEvaluationPublic: !settings.isEvaluationPublic,
-                                      },
-                                    },
-                                  })
-                                }}
-                                intl={intl}
-                                isEvaluationPublic={settings.isEvaluationPublic}
-                                runtime={runtime}
-                                sessionId={id}
-                                shortname={shortname}
-                                startedAt={dayjs(startedAt).format('HH:mm:ss')}
-                              />
+                                      Router.push('/sessions/running')
```

Comment:
There should be no redirect after a session block is reset. The necessary state should always be updated by Apollo automatically. If this is here for a state refresh, we will need to look at it in more detail.

Reply:
the redirect has been removed